src/pk/ecc/ecc_set_dp_internal.c

```diff
@@ -0,0 +1,150 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_MECC
+
+static void _ecc_oid_lookup(ecc_key *key)
+{
+   int err;
+   unsigned i;
+   void *tmp;
+   const ltc_ecc_curve *curve;
+
+   key->dp.oidlen = 0;
+   if ((err = mp_init(&tmp)) != CRYPT_OK) return;
+   for (curve = ltc_ecc_curves; curve->name != NULL; curve++) {
+      if ((err = mp_read_radix(tmp, curve->prime, 16)) != CRYPT_OK) continue;
```

Comment:
_copied from https://github.com/libtom/libtomcrypt/pull/236#discussion_r176715691_

how about converting the `key` elements once to strings and `strcmp()` then the elements of the sets?
alternatively add a function like
```
static int _ecc_cmp_hex_bn(const char* h, void* b, void* tmp)
{
   if (mp_read_radix(tmp, h, 16) != CRYPT_OK) return CRYPT_ERROR;
   if (mp_cmp(tmp, b) != LTC_MP_EQ) return CRYPT_ERROR;
   return CRYPT_OK;
}
```

Reply:
Well, I did basically the opposite thing, when loading the key I do convert all curve params (hex strings) into mpis. It saved couple of hex2mpi operations (grep for `mp_read_radix` in the old ecc implementation).

The function `_ecc_oid_lookup` is not heavily used, therefore I think an extra cost is acceptable here.

Reply:
Okay, perfectly fine. But please still add `_ecc_cmp_hex_bn()`